### ISSUE TITLE
return docs.count property of _cat/indices as a number

### ISSUE THREAD

**Issue by scampi** [elastic/elasticsearch#24091]
With the request `curl -XGET 'localhost:9200/_cat/indices/.kibi*?format=json'`, some fields are returned as strings but would make more sense to be returned as a number. Specifically, the fields are:

- docs.count
- docs.deleted
- pri/rep (primary/replicated ?)

See response of the previous request below:

```json
[
  {
    "health": "yellow",
    "status": "open",
    "index": "test",
    "uuid": "DW2GXroST0GpSdbsNnznCw",
    "pri": "1",
    "rep": "1",
    "docs.count": "121",
    "docs.deleted": "4",
    "store.size": "235.3kb",
    "pri.store.size": "235.3kb"
  }
]
```

I am using the ES instance below:

```json
{
  "name" : "x6CMBL4",
  "cluster_name" : "elasticsearch",
  "cluster_uuid" : "xU2OZXKBQRSLXpKEfyDYhA",
  "version" : {
    "number" : "5.2.2",
    "build_hash" : "f9d9b74",
    "build_date" : "2017-02-24T17:26:45.835Z",
    "build_snapshot" : false,
    "lucene_version" : "6.4.1"
  },
  "tagline" : "You Know, for Search"
}
```

**Comment by olcbean** [comment]
If this is an enhancement i would like to give it a try.

**Comment by jasontedor** [comment]
@olcbean This one requires a decision (whether or not we should do it), and care because it would be a breaking change.

**Comment by olcbean** [comment]
Oh, that's exactly what I meant by "if this is an enhancement" 😉

**Comment by jasontedor** [comment]
We discussed this during Fix-it-Friday. The non-cat APIs that provide the corresponding data here do provide these values as numerics and we think that that's what users should use if they need the data in such a format. We also discussed whether or not we should continue to support non-plain text output for the cat APIs. I will close this issue and open a issue for a separate discussion on that point.
